src/snowflake/cli/plugins/nativeapp/teardown_processor.py

```diff
@@ -103,30 +103,28 @@ def drop_application(
 
         # 3. Check if created by the Snowflake CLI
         row_comment = show_obj_row[COMMENT_COL]
-        if row_comment in ALLOWED_SPECIAL_COMMENTS:
-            # No confirmation needed before dropping
-            needs_confirm = False
```

Comment:
`needs_confirm` is never checked again, so we can just remove this whole block and invert the `if` check logic to make it cleaner